extensions/wikia/UserLogin/EmailConfirmationController.class.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+class EmailConfirmationController extends WikiaController {
+	/**
+	 * Attempt to confirm the user's email address and show success or failure
+	 */
+	public function postEmailConfirmation() {
+		$token = $this->wg->request->getVal( 'token', null );
+		$userFromToken = User::newFromConfirmationCode( $token );
+		$currentUserId = $this->wg->user->getId();
+
+		if (!$this->wg->request->wasPosted()) {
+			// Only POST to this resource is allowed
+			$this->response->setCode( 405 );
+		}
```

Comment:
This block should either return after setting the code, or this should be `elseif` after it, otherwise all the other checks will potentially overwrite the `wasPosted` check setting the 405 code and it could succeed anyway if it got through the other checks.